core/src/main/scala/org/apache/spark/rdd/BinaryShuffledRDD.scala

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.rdd
+
+import java.io.InputStream
+
+import scala.reflect.ClassTag
+
+import org.apache.spark.{BinaryShuffleDependency, Dependency, SparkEnv, TaskContext, Partition}
+import org.apache.spark.shuffle.BinaryShuffleWriter
+import org.apache.spark.storage.ShuffleBlockFetcherIterator
+
+/**
```

Comment:
Err, ignore my earlier comment and please take a look at this class as the entry point to understanding my proposal here.